src/platform/packages/shared/dashboards/dashboards-selector/components/dashboards_selector.tsx

```diff
@@ -7,25 +7,51 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 import { i18n } from '@kbn/i18n';
-import type { ContentManagementPublicStart } from '@kbn/content-management-plugin/public';
+import type { UiActionsPublicStart } from '@kbn/ui-actions-plugin/public';
+import type { ActionExecutionContext } from '@kbn/ui-actions-plugin/public';
 import type { EuiComboBoxOptionOption } from '@elastic/eui';
 import { EuiComboBox } from '@elastic/eui';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { debounce } from 'lodash';
-import { dashboardServiceProvider, type DashboardItem } from '../services/dashboard_service';
 
 interface DashboardOption {
   value: string;
   label: string;
 }
 
+interface Dashboard {
+  id: string;
+  isManaged: boolean;
+  title: string;
+}
+
+async function searchDashboards(
+  uiActions: UiActionsPublicStart,
+  search?: string,
+  perPage: number = 100
+): Promise<Dashboard[]> {
```

Comment:
I would probably pass an object here to avoid passing `undefined` to optional args.